src/suspendplugin.c

```diff
@@ -375,30 +365,93 @@ nl_get_multicast_id(
     err = 1;
 
     nl_cb_err(cb, NL_CB_CUSTOM, handle_nl_command_error, &err);
-    nl_cb_set(cb, NL_CB_ACK, NL_CB_CUSTOM, handle_nl_command_ack, &err);
     nl_cb_set(cb, NL_CB_VALID, NL_CB_CUSTOM, handle_nl_command_family, &grp);
+    nl_cb_set(cb, NL_CB_FINISH, NL_CB_CUSTOM, handle_nl_command_finished, &err);
+    nl_cb_set(cb, NL_CB_ACK, NL_CB_CUSTOM, handle_nl_command_ack, &err);
+    nl_cb_set(cb, NL_CB_SEQ_CHECK, NL_CB_CUSTOM, handle_nl_seq_check, &err);
 
     while (err > 0) {
-        nl_recvmsgs(sock, cb);
+        int res = 0;
+
+        res = nl_recvmsgs(sock, cb);
+        if (res < 0) {
+            connman_error("nl_recvmsgs failed - wmtWifi %s:%d\n", __func__, res);
+        }
     }
 
     if (err == 0) {
         err = grp.id;
     }
 
+
     nl_cb_put(cb);
     nlmsg_free(msg);
 
     return err;
 }
 
+static
+gboolean
+suspend_plugin_netlink_handler(
+    GIOChannel *channel,
+    GIOCondition cond,
+    gpointer data)
+{
+    struct nl_cb *cb;
+    int res = 0;
+    int err = 0;
+
+    if (cond & (G_IO_NVAL | G_IO_HUP | G_IO_ERR))
+        return FALSE;
+
+    cb = nl_cb_alloc(NL_CB_VERBOSE);
+    if (!cb)
+        return TRUE;
```

Comment:
Returning true without doing the io -> the callback will be triggered again -> if alloc fails this will turn into (silent) virtual busyloop. If the allocation failure is not really expected & retrying would be unlikely to remedy the situation, I'd disable the iowatch (with diagnostic message).

Reply:
or just don't check the returned value... With overcommit enabled, it's never going to return NULL anyway.